src/plugins/intel_gpu/src/graph/non_zero.cpp

```diff
@@ -27,6 +27,19 @@ layout count_nonzero_inst::calc_output_layout(count_nonzero_node const& node, ke
     return layout{cldnn::data_types::i32, cldnn::format::bfyx, tensor{1, 1, 1, 4}};
 }
 
+template<typename ShapeType>
+std::vector<layout> count_nonzero_inst::calc_output_layouts(count_nonzero_node const& /*node*/, kernel_impl_params const& impl_param) {
+    assert(static_cast<bool>(impl_param.desc->output_data_type) == false &&
+           "Output data type forcing is not supported for count_nonzero_node!");
+    if (impl_param.input_layouts[0].is_dynamic()) {
+        return { layout{ov::PartialShape{4}, cldnn::data_types::i32, cldnn::format::bfyx} };
+    } else {
+        auto rank =  static_cast<ov::Dimension::value_type>(impl_param.input_layouts[0].get_shape().size());
+        auto out_layout = layout{ov::PartialShape{rank + 1}, cldnn::data_types::i32, cldnn::format::bfyx};
```

Comment:
The logic here is not clear to me.
First of all, do we really need rank + 1 here? As I understand, you save OV_INPUT_RANK in addition to non-zero count, but it's not used in the kernel. And it's not necessary for gather_nonzero shape propagation as you can take rank from input layout there.

Secondly, don't we need a single output element (with non-zero elems count)? I mean can the it be simplified to something as follows w/o special branch for dynamic layouts?
```
std::vector<layout> count_nonzero_inst::calc_output_layouts(count_nonzero_node const& /*node*/, kernel_impl_params const& impl_param) {
    return { layout{ov::PartialShape{1}, cldnn::data_types::i32, cldnn::format::bfyx } };
}
```

Reply:
OV_INPUT_RANK is used in the kernel. 
(Previously it was used in both count and gather non_zero. However now I removed it from count nonzero) 

I like your suggestion to modify the count_non_zero result to be a single element of count value. 
I fixed according to that definition.